File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int left=0;
        int right=0;
        vector<int> v(26,0);
        int maxfreq=0;
        int maxlen=0;
        while(left<s.size()&&right<s.size())
        {
            v[s[right]-'A']++;
            maxfreq=max(maxfreq,v[s[right]-'A']);
            int windowsize=right-left+1;
            int replacement_chars=windowsize-maxfreq;
            if(replacement_chars<=k)
            {
                maxlen=max(maxlen,windowsize);
            }
            else
            {
                v[s[left]-'A']--;
                left++;
            }
             right++;
        }
        return maxlen;
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-1.py
class Solution:
    def trap(self, height: List[int]) -> int:
        left=[]
        for i in range(0,len(height)):
            if i==0:
                left.append(height[i])
            else:
                val=max(left[-1],height[i])
                left.append(val)
        right=[]
        for i in range(len(height)-1,-1,-1):
            if i==len(height)-1:
                right.append(height[i])
            else:
                val=max(height[i],right[-1])
                right.append(val)
        right.reverse()
        i=1
        totalwater=0
        for i in range(1,len(height)-1):
            val=min(left[i-1],right[i+1])-height[i]
            if val>0:
                totalwater+=val
        return totalwater


        